relay-event-normalization/src/normalize/span/description/mod.rs

```diff
@@ -193,51 +201,99 @@ fn scrub_file(description: &str) -> Option<String> {
     }
 }
 
-/// Normalize the domain and port of a URL.
+/// Scrub a `Url::Host` object. For domain names, run the string through a scrubber. For IP addresses, return the IP address as-is.
+///
+/// # Arguments
+///
+/// * `host` - `Url::Host`
+///
+/// # Returns
+///
+/// A host `String`, or `None` if scrubbing fails.
+pub fn scrub_host(host: Host<&str>) -> Cow<'static, str> {
+    match host {
+        Host::Ipv4(ip) => scrub_ipv4(ip),
+        Host::Ipv6(ip) => scrub_ipv6(ip),
+        Host::Domain(domain) => Cow::Owned(scrub_domain_name(String::from(domain))),
+    }
+}
+
+/// Scrub an IPV4 address. Allow well-known IPs like loopback, and fully scrub out all other IPs.
+///
+/// # Arguments
+///
+/// * `ip` - IP to be scrubbed
+///
+/// # Returns
+///
+/// Scrubbed string
+pub fn scrub_ipv4(ip: Ipv4Addr) -> Cow<'static, str> {
+    if IPV4_ALLOW_LIST.contains(&ip) {
+        return Cow::Owned(ip.to_string());
+    }
+
+    Cow::Borrowed("*.*.*.*")
+}
+
+/// Scrub an IPV6 address. Allow well-known IPs like loopback, and fully scrub out all other IPs.
+///
+/// # Arguments
+///
+/// * `ip` - IP to be scrubbed
+///
+/// # Returns
+///
+/// Scrubbed string
+pub fn scrub_ipv6(ip: Ipv6Addr) -> Cow<'static, str> {
+    if IPV6_ALLOW_LIST.contains(&ip) {
+        return Cow::Owned(ip.to_string());
+    }
+
+    Cow::Borrowed("*:*:*:*:*:*:*:*")
+}
+
+/// Sanitize a qualified domain string by replacing all but the last two segments with asterisks
 ///
 /// # Arguments
 ///
-/// * `domain` - The domain of the URL.
-/// * `port` - The port of the URL.
+/// * `domain` - Domain name (e.g., "my.domain.com")
 ///
 /// # Returns
 ///
-/// The normalized domain and port as a `String`, or `None` if normalization fails.
-pub fn normalize_domain(domain: &str, port: Option<u16>) -> Option<String> {
-    if let Some(allow_listed) = normalized_domain_from_allowlist(domain, port) {
-        return Some(allow_listed);
+/// The scrubbed domain as a `String`, or `None` if normalization fails.
+pub fn scrub_domain_name(domain: String) -> String {
```

Comment:
There is no need for this to always take a `String`.

```suggestion
pub fn scrub_domain_name(domain: &str) -> Cow<'_, str> {
```